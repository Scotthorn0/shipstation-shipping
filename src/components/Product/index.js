import React from 'react';
import { Header, Segment, Grid, Image } from 'semantic-ui-react';
import Ship from '../Ship';
import img from './bears-paw.jpg';
import './styles.css';

const plantDescription = `
  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
  Part sun to part shade. Nice house plant.`;

const Product = () => (
  <Segment>
    <Grid>
      <Grid.Row>
        <Grid.Column width={16}>
          <Header
            content="Order your amazing Succulent"
          />
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column width={4}>
          <Image size="large" src={img} />
        </Grid.Column>
        <Grid.Column width={8}>
          <Header content="Bear's Paw" />
          <Header sub content="Cotyledon ladismithensis" />
          <Segment basic>
            {plantDescription}
          </Segment>
        </Grid.Column>
        <Grid.Column width={4}>
          <Ship />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Segment>
);


export default Product;
