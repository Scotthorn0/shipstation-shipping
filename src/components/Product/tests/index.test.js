import React from 'react';
import { shallow } from 'enzyme';
import Product from '../';

describe('Product Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Product />);
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
