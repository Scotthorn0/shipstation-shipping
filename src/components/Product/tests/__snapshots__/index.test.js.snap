// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Grid>
        <GridRow>
                <GridColumn
                        width={16}
                >
                        <Header
                                content="Order your amazing Succulent"
                        />
                </GridColumn>
        </GridRow>
        <GridRow>
                <GridColumn
                        width={4}
                >
                        <Image
                                as="img"
                                size="large"
                                src="bears-paw.jpg"
                                ui={true}
                        />
                </GridColumn>
                <GridColumn
                        width={8}
                >
                        <Header
                                content="Bear's Paw"
                        />
                        <Header
                                content="Cotyledon ladismithensis"
                                sub={true}
                        />
                        <Segment
                                basic={true}
                        >
                                
                                  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                  Part sun to part shade. Nice house plant.
                        </Segment>
                </GridColumn>
                <GridColumn
                        width={4}
                >
                        <Ship />
                </GridColumn>
        </GridRow>
</Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridRow>
            <GridColumn
                        width={16}
            >
                        <Header
                                    content="Order your amazing Succulent"
                        />
            </GridColumn>
</GridRow>,
          <GridRow>
            <GridColumn
                        width={4}
            >
                        <Image
                                    as="img"
                                    size="large"
                                    src="bears-paw.jpg"
                                    ui={true}
                        />
            </GridColumn>
            <GridColumn
                        width={8}
            >
                        <Header
                                    content="Bear's Paw"
                        />
                        <Header
                                    content="Cotyledon ladismithensis"
                                    sub={true}
                        />
                        <Segment
                                    basic={true}
                        >
                                    
                                      Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                      Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                      Part sun to part shade. Nice house plant.
                        </Segment>
            </GridColumn>
            <GridColumn
                        width={4}
            >
                        <Ship />
            </GridColumn>
</GridRow>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridColumn
              width={16}
>
              <Header
                            content="Order your amazing Succulent"
              />
</GridColumn>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Header
                content="Order your amazing Succulent"
/>,
              "width": 16,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": "Order your amazing Succulent",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn
                width={4}
>
                <Image
                                as="img"
                                size="large"
                                src="bears-paw.jpg"
                                ui={true}
                />
</GridColumn>,
              <GridColumn
                width={8}
>
                <Header
                                content="Bear's Paw"
                />
                <Header
                                content="Cotyledon ladismithensis"
                                sub={true}
                />
                <Segment
                                basic={true}
                >
                                
                                  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                  Part sun to part shade. Nice house plant.
                </Segment>
</GridColumn>,
              <GridColumn
                width={4}
>
                <Ship />
</GridColumn>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  as="img"
                  size="large"
                  src="bears-paw.jpg"
                  ui={true}
/>,
                "width": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "img",
                  "size": "large",
                  "src": "bears-paw.jpg",
                  "ui": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Header
                    content="Bear's Paw"
/>,
                  <Header
                    content="Cotyledon ladismithensis"
                    sub={true}
/>,
                  <Segment
                    basic={true}
>
                    
                      Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                      Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                      Part sun to part shade. Nice house plant.
</Segment>,
                ],
                "width": 8,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "Bear's Paw",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "Cotyledon ladismithensis",
                    "sub": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "basic": true,
                    "children": "
  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
  Part sun to part shade. Nice house plant.",
                  },
                  "ref": null,
                  "rendered": "
  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
  Part sun to part shade. Nice house plant.",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Ship />,
                "width": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Grid>
          <GridRow>
                    <GridColumn
                              width={16}
                    >
                              <Header
                                        content="Order your amazing Succulent"
                              />
                    </GridColumn>
          </GridRow>
          <GridRow>
                    <GridColumn
                              width={4}
                    >
                              <Image
                                        as="img"
                                        size="large"
                                        src="bears-paw.jpg"
                                        ui={true}
                              />
                    </GridColumn>
                    <GridColumn
                              width={8}
                    >
                              <Header
                                        content="Bear's Paw"
                              />
                              <Header
                                        content="Cotyledon ladismithensis"
                                        sub={true}
                              />
                              <Segment
                                        basic={true}
                              >
                                        
                                          Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                          Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                          Part sun to part shade. Nice house plant.
                              </Segment>
                    </GridColumn>
                    <GridColumn
                              width={4}
                    >
                              <Ship />
                    </GridColumn>
          </GridRow>
</Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridRow>
              <GridColumn
                            width={16}
              >
                            <Header
                                          content="Order your amazing Succulent"
                            />
              </GridColumn>
</GridRow>,
            <GridRow>
              <GridColumn
                            width={4}
              >
                            <Image
                                          as="img"
                                          size="large"
                                          src="bears-paw.jpg"
                                          ui={true}
                            />
              </GridColumn>
              <GridColumn
                            width={8}
              >
                            <Header
                                          content="Bear's Paw"
                            />
                            <Header
                                          content="Cotyledon ladismithensis"
                                          sub={true}
                            />
                            <Segment
                                          basic={true}
                            >
                                          
                                            Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                            Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                            Part sun to part shade. Nice house plant.
                            </Segment>
              </GridColumn>
              <GridColumn
                            width={4}
              >
                            <Ship />
              </GridColumn>
</GridRow>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GridColumn
                width={16}
>
                <Header
                                content="Order your amazing Succulent"
                />
</GridColumn>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Header
                  content="Order your amazing Succulent"
/>,
                "width": 16,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "Order your amazing Succulent",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn
                  width={4}
>
                  <Image
                                    as="img"
                                    size="large"
                                    src="bears-paw.jpg"
                                    ui={true}
                  />
</GridColumn>,
                <GridColumn
                  width={8}
>
                  <Header
                                    content="Bear's Paw"
                  />
                  <Header
                                    content="Cotyledon ladismithensis"
                                    sub={true}
                  />
                  <Segment
                                    basic={true}
                  >
                                    
                                      Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                                      Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                                      Part sun to part shade. Nice house plant.
                  </Segment>
</GridColumn>,
                <GridColumn
                  width={4}
>
                  <Ship />
</GridColumn>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Image
                    as="img"
                    size="large"
                    src="bears-paw.jpg"
                    ui={true}
/>,
                  "width": 4,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "img",
                    "size": "large",
                    "src": "bears-paw.jpg",
                    "ui": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Header
                      content="Bear's Paw"
/>,
                    <Header
                      content="Cotyledon ladismithensis"
                      sub={true}
/>,
                    <Segment
                      basic={true}
>
                      
                        Cotyledon ladismithensis - Bear's Paw: Small shrublet.
                        Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
                        Part sun to part shade. Nice house plant.
</Segment>,
                  ],
                  "width": 8,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "content": "Bear's Paw",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "content": "Cotyledon ladismithensis",
                      "sub": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "basic": true,
                      "children": "
  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
  Part sun to part shade. Nice house plant.",
                    },
                    "ref": null,
                    "rendered": "
  Cotyledon ladismithensis - Bear's Paw: Small shrublet.
  Thick fuzzy leaves with toothed edge highlighted in dark red gives the appearance of a bear's paw.
  Part sun to part shade. Nice house plant.",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Ship />,
                  "width": 4,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
